Model/Observer/RecommendSettings.php

```diff
@@ -0,0 +1,216 @@
+            if (
+                $changedPath == $this->configHelper::IS_LOOKING_SIMILAR_ENABLED_IN_SHOPPING_CART
+                && $this->configHelper->isLookingSimilarEnabledInShoppingCart()
+            ) {
+                $this->validateLookingSimilar($changedPath);
+            }
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateFrequentlyBroughtTogether(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getBoughtTogetherRecommendation($this->getProductId());
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Frequently Bought Together recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateRelatedProducts(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getRelatedProductsRecommendation($this->getProductId());
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Related Products recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateTrendingItems(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getTrendingItemsRecommendation();
+            // When no recommendations suggested, most likely trained model is missing
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Trending Items recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
```

Comment:
Lots of redundancy here with the error handling. Seems we could abstract this into a more generic method?

Reply:
Respectfully disagree. Looks similar but not redundant. changedPath is different, different recommendations pulled, throws different error msg, try/catch and empty check is similar. We can try combine them together with switch case or others, but I feel like readability will be impacted.

Reply:
These things could be parameterized by the generic handler method no?

Reply:
Yes we can, then more if conditions to figure out which recommendations we are dealing with, call the right method and throw appropriate errors. I think breaking this into multiple observers will look better.